### ISSUE TITLE
Ternary conditional and null-coalescing operator in constructor results in faulty warning

### ISSUE THREAD

**Issue by trovialdo** [doxygen/doxygen#7358]
Doxygen generates a warning when the `a ? b : c` or ` a ?? b` is used in the parameters of the base constructor. This is a regression, it works in 1.8.14. This applies to C#.

Example
```
    /// <summary>
    /// Example 1
    /// </summary>
    /// <typeparam name="T">A param</typeparam>
    public class Derived<T> : Base<T>
    {
        /// <summary>
        /// Initialization
        /// </summary>
        /// <param name="value">My value, can be null</param>
        public Derived(List<T> value) : base(value is null ? new List<T>() : value)
        {

        }
    }
```
Faulty warning 1.8.16 `Class1.cs:30: warning: return type of member Demo.Derived.Derived? is not documented`
(there should be no warning)

Faulty warning 1.8.14 `Class1.cs:30: warning: parameters of member Demo.Derived.Derived are not (all) documented`
(i presume this has nothing to do with the `a ? b : c` or ` a ?? b`  operators but is another bug that has been fixed in the meanwhile?)

Expected no warnings.

MWE [Example.zip](https://github.com/doxygen/doxygen/files/3788923/Example.zip)

Tested on: 1.8.16 and 1.8.14

See documentation [ternary conditional operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/conditional-operator) and [null-coalescing operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator)
- `a ? b : c`  equals `if (a) { return b; } else { return c; }`
- `a ?? b` equals `if (a == null) { return b; } else { return a; }`

**Comment by trovialdo** [comment]
Updated to doxygen_config, what would be a more usual name?

**Comment by albert-github** [comment]
This name is OK the defautt name (and easiest to use as one can start doxygen without an argument) is Doxyfile

**Comment by albert-github** [comment]
I looked in the `Class1.cs`:
- on line 30 and 46 shouldn't `base` be `Base`? (so a capital B)
- on line 46 shouldn't `Derived` be `Derived2`? (so appending a 2)

Changing this doesn't solve the problem though.
Problem does not appear with 1.8.15 either

**Comment by trovialdo** [comment]
* line 30 is correct, base in this context means the constructor of the base class which, i unfortunately, also have called Base. Sorry for the confusion.
(you could also use `this` instead of `base` which would mean another constructor of your own, see below)

Example:
```
    public class Animal<T>
    {
        public Animal(List<T> value)        
       {       
       }
    }

    public class Horse<T> : Animal<T>
    {
        public Horse(List<T> value) : base(value is null ? new List<T>() : value)
        {
        }

      public Horse(int value) : this(value, 0)
      {
      }

      public Horse(int value1, int value2)
      {
      }
    }
```


* line 46, indeed typo on my end, corrected.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7364

**Comment by trovialdo** [comment]
Great thanks for the quick action, unfortunately i cannot test it since i have no box available to compile C++/doxygen on.

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
